component/org.wso2.carbon.identity.dpop/src/main/java/org/wso2/carbon/identity/dpop/validators/DPoPHeaderValidator.java

```diff
@@ -62,13 +62,23 @@ public class DPoPHeaderValidator {
      * @param tokReqMsgCtx Message context of token request.
      * @return DPoP header.
      */
-    public static String getDPoPHeader(OAuthTokenReqMessageContext tokReqMsgCtx) {
+    public static String getDPoPHeader(OAuthTokenReqMessageContext tokReqMsgCtx) throws IdentityOAuth2ClientException {
 
         HttpRequestHeader[] httpRequestHeaders = tokReqMsgCtx.getOauth2AccessTokenReqDTO().getHttpRequestHeaders();
         if (httpRequestHeaders != null) {
             for (HttpRequestHeader header : httpRequestHeaders) {
                 if (header != null && DPoPConstants.OAUTH_DPOP_HEADER.equalsIgnoreCase(header.getName())) {
-                    return ArrayUtils.isNotEmpty(header.getValue()) ? header.getValue()[0] : null;
+                    if (ArrayUtils.isNotEmpty(header.getValue())){
+                        if (header.getValue().length>1) {
+                            String error = "Multiple DPoP Proofs";
+                            if (log.isDebugEnabled()) {
+                                log.debug(error);
+                            }
+                            throw new IdentityOAuth2ClientException(DPoPConstants.INVALID_DPOP_PROOF, error);
+                        }
+                        return header.getValue()[0];
+                    }
+                    return null;
```

Comment:
do we handle NPE in the caller method?

Reply:
We can omit handling Null Pointer Exception (NPE) here since the output of this function is validated before usage using the following check: 
`if (StringUtils.isBlank(dPoPProof)){
}`.

This returns true for following cases:
 ```
  StringUtils.isBlank(null)      = true
  StringUtils.isBlank("")        = true
  StringUtils.isBlank(" ")       = true
```